packages/engine/src/worker/runner/javascript/hash_util.js

```diff
@@ -52,12 +52,47 @@
     return copy;
   };
 
+  // Workaround for regression tested at `units::state::multi_fixed_size`: Fixed size lists are not filled if set to
+  // `null`
+  //
+  // TODO: child-data for fixed-size-lists needs `vector.type.listSize` elements. This wrongly sets a `null` fixed-size-
+  //   list to `[null, null, ..., null]` to ensure this, but this behavior is wrong.
+  //   see https://app.asana.com/0/0/1201893568192898/f
+  //   see batch.js:array_data_from_col
+  //
+  // If `listSize` is set, we have a fixed size list. As we don't have offsets, the value must be set for `listSize`
+  // elements per field. If we have a `null` value, it gets replaced with `[null, null, ..., null]`.
+  const _fill_vector_for_fixed_size_list = (init, vector) => {
+    const size = vector.type.listSize;
+    if (init === null) {
+      if (vector.type.children[0].type.listSize) {
```

Comment:
I'm a bit worried about how specific this method is. What happens if the children type is a struct?

Reply:
I fixed this to check, if the type is a fixed-size-list